kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/registry/AbstractDeconstructingApplicationCommandRegistryStorage.kt

```diff
@@ -0,0 +1,79 @@
+package com.kotlindiscord.kord.extensions.registry
+
+import com.kotlindiscord.kord.extensions.commands.application.ApplicationCommand
+import dev.kord.common.entity.Snowflake
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.mapNotNull
+
+/**
+ * Abstract class which can be used to implement a simple networked key-value based registry storage
+ * for [ApplicationCommand]s.
+ *
+ * For simplicity the parameter / return types of the abstract methods are all [String]s.
+ */
+public abstract class AbstractDeconstructingApplicationCommandRegistryStorage<T : ApplicationCommand<*>> :
```

Comment:
You've got an abstract class here with a crapton of private and protected stuff. This all needs to be `public`, this is a framework we've got here.